isis/src/base/objs/Spice/Spice.cpp

```diff
@@ -170,41 +170,54 @@ namespace Isis {
     //  missions (e.g., MESSENGER) may augment the s/c SPK with new planet
     //  ephemerides. (2008-02-27 (KJB))
     if (m_usingNaif) {
-      std::string aleIsdStr = ale::load(cube.fileName().toStdString(), "", "isis");
-      
-      isd = json::parse(aleIsdStr);  
-      json aleNaifKeywords = isd["NaifKeywords"]; 
-      m_naifKeywords = new PvlObject("NaifKeywords", aleNaifKeywords); 
-
-      if (noTables) {
-        load(kernels["TargetPosition"], noTables);
-        load(kernels["InstrumentPosition"], noTables);
-        load(kernels["InstrumentPointing"], noTables);
-      }
-
-      if (kernels.hasKeyword("Frame")) {
-        load(kernels["Frame"], noTables);
-      }
+      try { 
+        // try using ALE
+        std::ostringstream kernel_pvl;
+        kernel_pvl << kernels;
+
+        json props;
+        props["kernels"] = kernel_pvl.str();
+
+        isd = json::parse(ale::loads(cube.fileName().toStdString(), props.dump(), "isis"));
+        json aleNaifKeywords = isd["NaifKeywords"];
+        m_naifKeywords = new PvlObject("NaifKeywords", aleNaifKeywords);
```

Comment:
Does this expect the NaifKeywords from ALE to be a PVL formatted string embedded in the JSON object?

Reply:
no, PvlObject has a constructor that converts the json format to PVL. So the JSON is the standard json object in ALE and the returned PVL is formatted exactly as ISIS expects it to be.